### ISSUE TITLE
Migrate away from deprecated `Pads`/`get_pads` mechanism

### ISSUE THREAD

**Issue by whitequark** [GlasgowEmbedded/glasgow#599]
The two mechanisms are similar conceptually and mainly differ syntactically. The following command can illustrate the changes required:

```console
$ git diff 38730032..e9571705 -- glasgow/applet/interface/jtag_probe/__init__.py
```

List of applets that still need migration:

- [x] applet.audio.dac
- [x] applet.audio.yamaha_opx
- [x] applet.control.servo
- [x] applet.display.hd44780
- [x] applet.display.pdi
- [x] applet.interface.analyzer
- [x] applet.interface.i2c_initiator
- [x] applet.interface.i2c_target
- [x] applet.interface.ps2_host
- [x] applet.interface.sbw_probe
- [x] applet.interface.spi_controller
- [x] applet.interface.swd_openocd
- [x] applet.interface.uart
- [x] applet.memory.floppy
- [x] applet.memory.onfi
- [x] applet.memory.prom
- [x] applet.program.avr.spi
- [x] applet.program.m16c
- [x] applet.program.nrf24lx1
- [x] applet.radio.nrf24l01
- [x] applet.sensor.hx711
- [x] applet.sensor.pmsx003
- [x] applet.video.hub75_output
- [x] applet.video.rgb_input
- [x] applet.video.vga_output
- [x] boilerplate applet
- [x] applet.interface.jtag_pinout
- [x] applet.interface.spi_flashrom
- [x] applet.program.ice40_flash

**Comment by q3k** [comment]
Like this? (#602) My Amaranth is very rusty and I don't have how to actually test this. Is there an easy way to make sure an Amaranth/Python change is an RTL no-op?

**Comment by whitequark** [comment]
> Is there an easy way to make sure an Amaranth/Python change is an RTL no-op?

No, but the changes are simple enough I expect to catch most issues during review, even if it's tedious.

Improvements to the general applet infrastructure will continue at a decent pace for the next half year or so, and I expect I'll have to sit down and manually test every single applet at the end of it, hopefully while also adding some better tests. It's going to be painful but what can you do?..

**Comment by whitequark** [comment]
Also I don't even know where *my* HX711 board is, much less have a load cell on hand...

**Comment by q3k** [comment]
Any tips on sequencing updates across multiple modules, eg. with UART(pads/ports) being used by multiple applets?

Is there a temporary/transitional mechanism that I could use to first update the applets (or libraries?) and then the other side?

**Comment by whitequark** [comment]
I'd suggest leaving updating the libraries (UART and I2C in particular) to some later point, they're kind of a pain and I'm going to rewrite UART at least soon, anyway.

For the rest, unfortunately we don't have real applet composability so the answer is suffering. Making things composable is on my TODO list but it goes after this.

**Comment by isabelburgos** [comment]
One more item for this list - `examples.boilerplate` still uses pads
